src/main/java/com/amazon/opendistroforelasticsearch/security/privileges/Evaluator.java

```diff
@@ -0,0 +1,35 @@
+package com.amazon.opendistroforelasticsearch.security.privileges;
+
+import com.amazon.opendistroforelasticsearch.security.configuration.ConfigurationChangeListener;
+import com.amazon.opendistroforelasticsearch.security.user.User;
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.common.transport.TransportAddress;
+import org.elasticsearch.tasks.Task;
+
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * @author Divyansh Jain
+ */
+
+public interface Evaluator extends ConfigurationChangeListener {
+
+    EvaluatorResponse evaluate(final User user, String action0, final ActionRequest request, Task task);
+
+    boolean isInitialized();
+
+    Set<String> mapRoles(User user, TransportAddress remoteAddress);
+
+    Map<String, Boolean> mapTenants(User user, Set<String> securityRoles);
+
+    boolean notFailOnForbiddenEnabled();
+
+    boolean multitenancyEnabled();
+
+    String kibanaIndex();
+
+    String kibanaServerUsername();
+
+    Set<String> getAllConfiguredTenantNames();
```

Comment:
Do we need these APIs in this interface definition ? If both default implementation of Privileges Evaluator and Ranger implementation are using the same code for these APIs then we can add these API defnitions in abstract class 'AbstractEvaluator' only.